### PR TITLE
added partontop on dileptoncommon

### PR DESCRIPTION
**Pull request by tt8888tt**
added gentop on dilepton but need to be checked...
TTdilepDraw errorband is corrected

### CODE REVIEW

File: CatAnalyzer/test/run_TtbarDiLeptonAnalyzer_cfg.py [hunk 1]
```diff
@@ -9,7 +9,8 @@
 
 process.source = cms.Source("PoolSource", fileNames = cms.untracked.vstring())
 from CATTools.Validation.commonTestInput_cff import commonTestCATTuples
-process.source.fileNames = commonTestCATTuples["sig"]
+process.source.fileNames = ["root://cms-xrdr.sdfarm.kr:1094///xrd/store/group/CAT/TT_TuneCUETP8M2T4_13TeV-powheg-pythia8/v8-0-4_RunIISummer16MiniAODv2-PUMoriond17_80X_mcRun2_asymptotic_2016_TrancheIV_v6-v1/170113_045423/0000/catTuple_983.root"]
+#process.source.fileNames = commonTestCATTuples["sig"]
```
Comment (jhgoh) [inline CatAnalyzer/test/run_TtbarDiLeptonAnalyzer_cfg.py]: My idea here was to use the common root file for the testing. Is there any special reason to use this root file? To pick up v804?

File: CatAnalyzer/plugins/dileptonCommon.cc [hunk 2]
```diff
@@ -294,84 +300,116 @@ int dileptonCommon::eventSelection(const edm::Event& iEvent, const edm::EventSet
...
-      b_gen_partonMode1 = 0;
-      b_gen_partonMode2 = 0;
-    }
-    else if ( (*partonTop_modes).size() == 1 ) { b_gen_partonMode2 = 0; }
-    else{
-      b_gen_partonChannel = *partonTop_channel;
-      b_gen_partonMode1 = (*partonTop_modes)[0];
-      b_gen_partonMode2 = (*partonTop_modes)[1];
+    edm::Handle<vector<reco::GenParticle> > genParticleHandle;
+    iEvent.getByToken(partonTop_, genParticleHandle);
```
Comment (jhgoh) [inline CatAnalyzer/plugins/dileptonCommon.cc]: Looks like you are redoing the partonTopProducer. Is there any problem to use it? Maybe you can update the PartonTopProducer, not here if you observe any problem or suggestion to improve it.

File: CatAnalyzer/test/run_TtbarDiLeptonAnalyzer_cfg.py (hunk 1 above)
Comment (tt8888tt) [reply CatAnalyzer/test/run_TtbarDiLeptonAnalyzer_cfg.py]: Nope i think its genuine mistake
Supposed to not to be commited

File: CatAnalyzer/plugins/dileptonCommon.cc (hunk 2 above)
Comment (tt8888tt) [reply CatAnalyzer/plugins/dileptonCommon.cc]: Yeah i just didnt know how to technically
Im getting figure it out

**Comment by tt8888tt** [comment]
Seems it was too premature pull reqeust
Maybe i have to revet these commits... maybe should close this pr?

**Comment by jshlee** [comment]
u can just fix this one

**Comment by jhgoh** [comment]
Ready to be merged?

**Comment by tt8888tt** [comment]
Yes

**Pull request merged**